webpack.config.js

```diff
@@ -6,19 +6,53 @@ const OptimizeCssAssetsPlugin = require('optimize-css-assets-webpack-plugin');
 const UglifyJsPlugin = require('uglifyjs-webpack-plugin');
 const UnminifiedWebpackPlugin = require('unminified-webpack-plugin');
 const ThreadsPlugin = require('threads-plugin');
+const WebpackBeforeBuildPlugin = require('before-build-webpack');
+const fs = require('fs');
 
-module.exports = {
+// the WaitPlugin is copypasta from:
+// https://www.viget.com/articles/run-multiple-webpack-configs-sequentially/
+class WaitPlugin extends WebpackBeforeBuildPlugin {
+  constructor(file, interval = 100, timeout = 10000) {
+    super(function (stats, callback) {
+      let start = Date.now();
+
+      function poll() {
+        if (fs.existsSync(file)) {
+          callback();
+        } else if (Date.now() - start > timeout) {
+          throw Error("Maybe it just wasn't meant to be.");
+        } else {
+          setTimeout(poll, interval);
+        }
+      }
+
+      poll();
+    });
+  }
+}
+
+const workerConfig = {
+  output: {
+    filename: 'worker.js',
+    path: path.resolve(__dirname, 'dist'),
+  },
+  entry: path.resolve(__dirname, 'src/bam-fetcher-worker'),
+  target: 'webworker',
+  plugins: [new UnminifiedWebpackPlugin(), new ThreadsPlugin()],
+};
+
+const libraryConfig = {
   output: {
     filename: 'higlass-pileup.min.js',
     library: 'higlass-pileup',
     libraryTarget: 'umd',
     path: path.resolve(__dirname, 'dist'),
   },
   devServer: {
-    contentBase: [
-      path.join(__dirname, 'node_modules/higlass/dist'),
-    ],
+    contentBase: [path.join(__dirname, 'node_modules/higlass/dist')],
     watchContentBase: true,
+    port: 8077,
+    writeToDisk: true,
```

Comment:
Are these changes necessary?

Reply:
The port one isn't. The write to disk is because we need it for the dependency resolution.